kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -262,6 +267,17 @@ protected TransactionImpl buildTransactionInternal(
     if (isCreateOrReplace) {
       // For a new table definition start with an empty initial metadata
       baseMetadata = getInitialMetadata();
+      // In the case of Replace table there are a few delta-specific properties we want to preserve
+      if (latestSnapshot.isPresent()) {
```

Comment:
So the semantic/assumption/conclusion here is that:

If it is a replace table operation then we expect `isLatestSnapshot.isPresent` to also be true, right? i.e. that's how we are concluding that it is a replace table operation in the first place?

Worth adding a comment explaining that?

Reply:
Added a comment to hopefully clarify